load/dependencies.go

```diff
@@ -9,13 +9,14 @@ import (
 
 type TxGenerator[T comparable] interface {
 	// GenerateTx returns a valid transaction.
-	GenerateTx(ctx context.Context) (T, error)
+	GenerateTx() (T, error)
 }
 
 type Issuer[T comparable] interface {
 	// Listen for the final status of transactions and notify the tracker
 	// Listen stops if the context is done, an error occurs, or if the issuer
 	// has sent all their transactions.
+	// Listen MUST return a nil error if the context is canceled.
```

Comment:
Shouldn't this be the opposite?? If the context is canceled, we usually reached the final timeout without collecting all the transactions back, that seems like we should error?